src/test/javascript/spec/component/shared/metis/post/post.component.spec.ts

```diff
@@ -1,53 +1,96 @@
-import * as chai from 'chai';
-import sinonChai from 'sinon-chai';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { MockComponent, MockPipe } from 'ng-mocks';
-import { DebugElement } from '@angular/core';
+import { MockComponent, MockDirective, MockPipe } from 'ng-mocks';
+import { DebugElement, Directive, Input } from '@angular/core';
 import { HtmlForMarkdownPipe } from 'app/shared/pipes/html-for-markdown.pipe';
 import { PostComponent } from 'app/shared/metis/post/post.component';
 import { getElement } from '../../../../helpers/utils/general.utils';
 import { PostFooterComponent } from 'app/shared/metis/postings-footer/post-footer/post-footer.component';
 import { PostHeaderComponent } from 'app/shared/metis/postings-header/post-header/post-header.component';
 import { metisPostExerciseUser1 } from '../../../../helpers/sample/metis-sample-data';
+import { PostingContentComponent } from 'app/shared/metis/posting-content/posting-content.components';
+import { MockMetisService } from '../../../../helpers/mocks/service/mock-metis-service.service';
+import { MetisService } from 'app/shared/metis/metis.service';
 
-chai.use(sinonChai);
-const expect = chai.expect;
+// tslint:disable-next-line:directive-selector
+@Directive({ selector: '[routerLink]' })
+export class MockRouterLinkDirective {
+    @Input('routerLink') data: any;
+}
+
+// tslint:disable-next-line:directive-selector
+@Directive({ selector: '[queryParams]' })
+export class MockQueryParamsDirective {
+    @Input('queryParams') data: any;
+}
 
 describe('PostComponent', () => {
     let component: PostComponent;
     let fixture: ComponentFixture<PostComponent>;
     let debugElement: DebugElement;
+    let metisService: MetisService;
+    let metisServiceGetLinkMock: jest.SpyInstance;
+    let metisServiceGetQueryParamsMock: jest.SpyInstance;
 
     beforeEach(() => {
         return TestBed.configureTestingModule({
-            declarations: [PostComponent, MockPipe(HtmlForMarkdownPipe), MockComponent(PostHeaderComponent), MockComponent(PostFooterComponent)],
+            providers: [{ provide: MetisService, useClass: MockMetisService }],
+            declarations: [
+                PostComponent,
+                MockPipe(HtmlForMarkdownPipe),
+                MockComponent(PostHeaderComponent),
+                MockComponent(PostingContentComponent),
+                MockComponent(PostFooterComponent),
+                MockDirective(MockRouterLinkDirective),
```

Comment:
already changed in c80c3e6, thanks for bringing it up

Reply:
Ah, I was too slow. Personally I would have preferred the use of MockDirective, since it does not require the redefinition of the deirectives, but either is fine I guess.